### PR TITLE
Turn storage provider into a module + api to re-hydrate upon refresh

### PR DESCRIPTION
**Pull request by andrewvmail**
An improvement to avoid getter setter ```localStorage``` boilercodes in the controller options. 

Referencing #1064

### CODE REVIEW

**Comment by henri-hulski** [comment]
Please use `npm run commit` so the commit message would be standardized for rendering the release notes. 

And remember to rebase to next. :-)

**Comment by andrewvmail** [comment]
@henri-hulski  oops i didnt read your message before committing the first 2 commit. Although i tried to make it as similar as you everyone elses. I will do run commit next one. Thanks for the heads up!

**Comment by henri-hulski** [comment]
Could you rename the first commit according to commitizen?

**Comment by andrewvmail** [comment]
@henri-hulski done, i think im getting this rebase thing

**Comment by henri-hulski** [comment]
Great! 🥇

File: packages/node_modules/cerebral/src/BaseController.js [hunk 1]
```diff
@@ -50,6 +50,8 @@ class BaseController extends FunctionTree {
     })
     this.model = new config.Model(this)
 
+    this.emit('initialized:model')
```
Comment (Guria) [inline packages/node_modules/cerebral/src/BaseController.js]: @christianalfoni are you ok with new event?

if yes we should also document it in api/controller

Comment (christianalfoni) [reply packages/node_modules/cerebral/src/BaseController.js]: Jup, discussed this beforehand :)

**Comment by christianalfoni** [comment]
Okay, implemented this into PR: https://github.com/cerebral/cerebral/pull/1134

**Pull request closed without merging**